src/SwedbankPay.Sdk/PaymentInstruments/Invoice/InvoicePayment.cs

```diff
@@ -0,0 +1,12 @@
+﻿namespace SwedbankPay.Sdk.PaymentInstruments.Invoice
+{
+    public class InvoicePayment : IInvoicePayment
+    {
+        public InvoicePayment(InvoiceType invoiceType)
+        {
+            InvoiceType = invoiceType;
+        }
+
+        public InvoiceType InvoiceType { get; set; }
```

Comment:
If `InvoiceType` should have a public setter, I don't think we need it in the constructor. If this class is used for requests, we should just ensure that the experience is as simple as possible in order to create the necessary request objects. I think we still need to do some refactoring to optimize in that area. There are too many nullable parameters and such in use currently.

```suggestion
        public InvoiceType InvoiceType { get; set; }
```